### PR TITLE
[KOGITO-3874] - Allow configuration of readiness and liveness probes

### PR DESCRIPTION
**Pull request by vaibhavjainwiz**
Many thanks for submitting your Pull Request :heart:! 

Jira issue : https://issues.redhat.com/browse/KOGITO-3874

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-cloud-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains a link to the JIRA issue
- [x] Pull Request contains a description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: pkg/apis/app/v1beta1/probe.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+// KogitoProbe describes a health check to be performed against a container to determine whether it is
+// alive or ready to receive traffic.
+// +k8s:openapi-gen=true
+type KogitoProbe struct {
```
Comment (ricardozanini) [inline pkg/apis/app/v1beta1/probe.go:20]: Please use the Kubernetes probe object instead.

**Review by ricardozanini — CHANGES_REQUESTED** [review]
Hey @vaibhavjainwiz! Thanks for this PR. I'd say that we can use the probe API from Kubernetes core instead of cloning its attributes.

File: pkg/apis/app/v1beta1/probe.go (hunk 1 above)
Comment (vaibhavjainwiz) [reply pkg/apis/app/v1beta1/probe.go]: done

**Review by spolti — CHANGES_REQUESTED** [review]

File: pkg/infrastructure/services/probe.go [hunk 2]
```diff
@@ -40,66 +41,61 @@ type healthCheckProbe struct {
...
 // getProbeForKogitoService gets the appropriate liveness (index 0) and readiness (index 1) probes based on the given service definition
-func getProbeForKogitoService(serviceDefinition ServiceDefinition) healthCheckProbe {
+func getProbeForKogitoService(serviceDefinition ServiceDefinition, service v1beta1.KogitoService) healthCheckProbe {
 	switch serviceDefinition.HealthCheckProbe {
 	case QuarkusHealthCheckProbe:
 		return healthCheckProbe{
-			readiness: getQuarkusHealthCheckReadiness(),
-			liveness:  getQuarkusHealthCheckLiveness(),
+			readiness: getQuarkusHealthCheckReadiness(service.GetSpec().GetReadinessProbe()),
+			liveness:  getQuarkusHealthCheckLiveness(service.GetSpec().GetLivenessProbe()),
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply pkg/infrastructure/services/probe.go:49]: I believe we have, and I think we have a JIRA files for this. If not, we must add it.

**Review by ricardozanini — APPROVED** [review]

**Review by spolti — APPROVED** [review]

**Comment by sutaakar** [comment]
Should we provide a support in CLI too?

**Comment by sutaakar** [comment]
Otherwise works as expected

**Comment by ricardozanini** [comment]
> Should we provide a support in CLI too?

This fine tune can be done by editing the resource after being deployed. It's an edge case. What we could do is to provide some parameters in the CLI, not the entire probe object. Like the 'initial delay seconds', for example.

**Review by sutaakar — APPROVED** [review]

**Review by Kaitou786 — APPROVED** [review]

**Comment by ricardozanini** [comment]
@sutaakar merging, but would be nice to have a JIRA to follow up your suggestion.

**Comment by sutaakar** [comment]
Created https://issues.redhat.com/browse/KOGITO-3898 to track the CLI changes.

**Pull request merged**